Signal/src/ViewControllers/HomeView/HomeViewController.m

```diff
@@ -197,12 +211,29 @@ - (void)loadView
         [SignalApp.sharedApp setHomeViewController:self];
     }
 
+    __weak HomeViewController *weakSelf = self;
+    ReminderView *deregisteredView =
+        [ReminderView nagWithText:NSLocalizedString(@"DEREGISTRATION_WARNING",
+                                      @"Label warning the user that they have been de-registered.")
+                        tapAction:^{
+                            HomeViewController *strongSelf = weakSelf;
+                            if (!strongSelf) {
+                                return;
+                            }
+                            [RegistrationUtils showReregistrationUIFromViewController:strongSelf];
+                        }];
+    [self.view addSubview:deregisteredView];
+    [deregisteredView autoPinWidthToSuperview];
+    [deregisteredView autoPinToTopLayoutGuideOfViewController:self withInset:0];
+    self.hideDeregisteredViewConstraint = [deregisteredView autoSetDimension:ALDimensionHeight toSize:0];
```

Comment:
I understand you're building on the existing pattern, but it's kind of gnarly. Could we instead put all the reminder views into a (potentially empty) reminderViewStackView and remove the `hide*ViewConstraint`s?

It's not a big deal and this should work as is, but I think the alternative would be cleaner. TIOLI

Reply:
Done.

Reply:
That is, I kept the constraints but used a stack view.  If you'd rather go farther and rework this to not use the constraints, we could do that but I'm not sure it'll be any simpler.